plugins/catalog/src/components/HasResourcesCard/HasResourcesCard.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RELATION_HAS_PART } from '@backstage/catalog-model';
+import React from 'react';
+import {
+  asResourceEntities,
+  RelatedEntitiesCard,
+  resourceEntityColumns,
+  resourceEntityHelpLink,
+} from '../RelatedEntitiesCard';
+
+type Props = {
+  variant?: 'gridItem';
+};
+
+export const HasResourcesCard = ({ variant = 'gridItem' }: Props) => {
+  return (
+    <RelatedEntitiesCard
+      variant={variant}
+      title="Resources"
+      entityKind="Resource"
+      relationType={RELATION_HAS_PART}
+      columns={resourceEntityColumns}
+      asRenderableEntities={asResourceEntities}
+      emptyMessage="No resource is part of this system"
```

Comment:
just noting that these cards make assumptions about what the _source_ of the relation is too. That may reduce their utility.

Would it make sense to grab the entity the relations start from, and make the text something like ``` This ${entity.kind.toLocaleLowercase('en-US')} has no associated resources ``` or something like that?

Reply:
For sure.  On the one hand, the initial versions of these cards did too, preferring the semantic to be "sort of" encoded in the relationship.  I tried to follow the same semantic in naming, and as it turned out, HasResources only applies to a system.  HasComponents *could* apply to a system or to a component (via subcomponents), but there were both HasComponents and HasSubcomponents defined.  It gets even more confusing when adding cards for DependsOnComponent.  All of these cards do the same thing, and only this message or title varies.

Reply:
I'm pretty much open to suggestions here, but changing from the `RelationshipnameTargetkindCard` to the `SourcekindRelationpurposeMaybetargetkindCard` semantic will introduce breaking changes for apps relying on the existing `HasSystemsCard`, `HasComponentsCard`, and `HasSubcomponentsCard`.

What feels like the right approach to you?